api/v1alpha1/common_types.go

```diff
@@ -75,6 +76,12 @@ type Record struct {
 	Id          string `json:"id"`
 	SelectorKey string `json:"selectorKey"`
 	Phase       Phase  `json:"phase"`
+	// InjectedCount is a counter to record the sum of successful injections
+	InjectedCount int `json:"injectedCount"`
+	// RecoveredCount is a counter to record the sum of successful recoveries
+	RecoveredCount int `json:"recoveredCount"`
+	// Events are the essential details about the injections and recoveries
+	Events []*RecordEvent `json:"events,omitempty"`
```

Comment:
```suggestion
	// Events are the essential details about the injections and recoveries
	Events []RecordEvent `json:"events,omitempty"`
```

I think the element in this slice could not be `nil`, and we would not update the element, so we do not require to use "slice of pointers".